### PR TITLE
Added parseaddr and removed some Exception aliases that were never called

### PR DESCRIPTION
**Pull request by bwelboren**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `flake8`
- [x] `python manage.py test`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: api/modules/city/views.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def get_city(request, city_id):
     try:
         city_visit_log = CityVisitLog(city=city, user=request.user)
         city_visit_log.save()
-    except Exception as e:
+    except Exception:
```
Comment (prabhakar267) [inline api/modules/city/views.py]: Please do not add files which are not relevant to the fix.
If you want to remove these unused alias, you are raise another PR

File: api/modules/users/validators.py [hunk 2]
```diff
@@ -13,7 +15,7 @@ def _has_at(email):
     :param email:
     :return: boolean
     """
-    return "@" in email
+    return '@' in parseaddr(email)[1]
```
Comment (prabhakar267) [inline api/modules/users/validators.py]: Do not modify existing validators, instead add a new one.
Also add tests for the same

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

**Comment by bwelboren** [comment]
Okay I added a new function and ran a test on it

File: api/modules/city/views.py (hunk 1 above)
Comment (Swati4star) [reply api/modules/city/views.py]: @bjornwelboren Please remove this file from commit as there is no relevant change here.
Do the same for the other 2 files also.

File: api/modules/currency/views.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def get_currency_exchange_rate(request, source_currency_code, target_currency_co
                                 target=target_currency_code,
                                 result=api_response_json[query])
 
-    except Exception as e:
+    except Exception:
```
Comment (Swati4star) [inline api/modules/currency/views.py:35]: @bjornwelboren Don't delete the files. They just shouldnt be a part of this commit

**Pull request closed without merging**